test/jdk/tools/jar/modularJar/Basic.java

```diff
@@ -942,6 +950,127 @@ public void updateFooModuleVersion() throws IOException {
+            .map(ModuleReferenceImpl.class::cast)
+            .orElseThrow();
+        ModuleResolution moduleResolution = moduleReference.moduleResolution();
+
+        assertTrue(hasWarning.test(moduleResolution));
+    }
+
+    /**
+     * Validate that you can create a jar only specifying --do-not-resolve-by-default
+     * @throws IOException
+     */
+    @Test
+    public void updateFooModuleResolutionWarnIfResolved() throws IOException {
+        Path mp = Paths.get("updateFooModuleResolutionDoNotResolveByDefault");
+        createTestDir(mp);
+        Path modClasses = MODULE_CLASSES.resolve(FOO.moduleName);
+        Path modularJar = mp.resolve(FOO.moduleName + ".jar");
+
+        jar("--create",
+            "--file=" + modularJar.toString(),
+            "--main-class=" + FOO.mainClass,
+            "--do-not-resolve-by-default",
+            "--no-manifest",
+            "-C", modClasses.toString(), ".")
+            .assertSuccess();
+
+        ModuleReferenceImpl moduleReference = ModuleFinder.of(modularJar).find(FOO.moduleName)
+            .map(ModuleReferenceImpl.class::cast)
+            .orElseThrow();
+        ModuleResolution moduleResolution = moduleReference.moduleResolution();
+
+        assertTrue(moduleResolution.doNotResolveByDefault());
+    }
+
+    /**
+     * Validate that you can create a jar specifying --warn-if-resolved and --do-not-resolve-by-default
+     * @throws IOException
+     */
+    @Test(dataProvider = "resolutionNames")
+    public void updateFooModuleResolutionWarnIfResolvedAndDoNotResolveByDefault(String resolutionName, Predicate<ModuleResolution> hasWarning) throws IOException {
+        Path mp = Paths.get("updateFooModuleResolutionWarnThenNotResolve-" + resolutionName);
+        createTestDir(mp);
+        Path modClasses = MODULE_CLASSES.resolve(FOO.moduleName);
+        Path modularJar = mp.resolve(FOO.moduleName + ".jar");
+
+        jar("--create",
+            "--file=" + modularJar.toString(),
+            "--main-class=" + FOO.mainClass,
+            "--warn-if-resolved=" + resolutionName,
+            "--do-not-resolve-by-default",
+            "--no-manifest",
+            "-C", modClasses.toString(), ".")
+            .assertSuccess();
+
+        ModuleReferenceImpl moduleReference = ModuleFinder.of(modularJar).find(FOO.moduleName)
+            .map(ModuleReferenceImpl.class::cast)
+            .orElseThrow();
+        ModuleResolution moduleResolution = moduleReference.moduleResolution();
+
+        assertTrue(hasWarning.test(moduleResolution));
+        assertTrue(moduleResolution.doNotResolveByDefault());
+    }
+
+    /**
+     * Validate that you can create a jar specifying --do-not-resolve-by-default and --warn-if-resolved
+     * @throws IOException
+     */
+    @Test(dataProvider = "resolutionNames")
+    public void updateFooModuleResolutionDoNotResolveByDefaultAndWarnIfResolved(String resolutionName, Predicate<ModuleResolution> hasWarning) throws IOException {
```

Comment:
Update the existing tests is good but overly long lines make it too hard to read (it will make future side-by-side view impossible). Can you trim all these lines down to something closer to the existing tests in this file?